code/game/mecha/mecha.dm

```diff
@@ -400,6 +417,14 @@
 	diag_hud_set_mechstat()
 	diag_hud_set_mechtracking()
 
+	//Check if we should ignite the pilot of an open-canopy mech
+	var/turf/target = get_turf(src)

```

Comment:
This should be in fire_act or similar instead of process.

Reply:
The fire_act() proc isn't available for mechs - it's a mob-only proc, so I can't use it to check for fire.

Reply:
fire_act is atom proc.

Reply:
Huh, dunno what I did wrong last time I tried it, but I got it working now. Will push an update out soon.